src/Form/DataTransformer/ServiceProviderDataTransformer.php

```diff
@@ -49,7 +48,7 @@ public function reverseTransform($value)
         try {
             $this->provider->transform($value);
         } catch (\Throwable $e) {
-            \assert(null !== $this->logger);
+            $this->logger = $this->logger ?? new NullLogger();
```

Comment:
This looks strange. Is this symfony new best practice?

There's a risk that some other method might use the logger too.

Reply:
Yes, why not keeping the old way ?

Reply:
Fixed using local variables.

Reply:
> I think it's because the `LoggerAwareTrait` allow the logger to be null.

I think we should remove the trait and add a dedicated property that can't be null to fix this problem.